pyanaconda/storage/kickstart.py

```diff
@@ -0,0 +1,199 @@
+
+
+def _update_autopart(storage):
+    """Update data for automatic partitioning.
+
+    :param storage: an instance of the storage
+    """
+    auto_part_proxy = STORAGE.get_proxy(AUTO_PARTITIONING)
+    auto_part_proxy.SetEnabled(storage.do_autopart)
+    auto_part_proxy.SetType(storage.autopart_type)
+    auto_part_proxy.SetEncrypted(storage.encrypted_autopart)
+
+    if storage.encrypted_autopart:
+        auto_part_proxy.SetLUKSVersion(storage.autopart_luks_version)
+
+        if storage.autopart_pbkdf_args:
+            auto_part_proxy.SetPBKDF(storage.autopart_pbkdf_args.type or "")
+            auto_part_proxy.SetPBKDFMemory(storage.autopart_pbkdf_args.max_memory_kb)
+            auto_part_proxy.SetPBKDFIterations(storage.autopart_pbkdf_args.iterations)
+            auto_part_proxy.SetPBKDFTime(storage.autopart_pbkdf_args.time_ms)
+
+
+def _update_clearpart(storage):
+    """Update data for clearpart.
+
+    :param storage: an instance of the storage
+    """
+    disk_init_proxy = STORAGE.get_proxy(DISK_INITIALIZATION)
+    disk_init_proxy.SetInitializationMode(storage.config.clear_part_type)
+    disk_init_proxy.SetDrivesToClear(storage.config.clear_part_disks)
+    disk_init_proxy.SetDevicesToClear(storage.config.clear_part_devices)
+    disk_init_proxy.SetInitializeLabelsEnabled(storage.config.initialize_disks)
+
+    if disk_init_proxy.InitializationMode == CLEAR_PARTITIONS_NONE:
+        # Make a list of initialized disks and of removed partitions. If any
+        # partitions were removed from disks that were not completely
+        # cleared we'll have to use CLEAR_PARTITIONS_LIST and provide a list
+        # of all removed partitions. If no partitions were removed from a
+        # disk that was not cleared/reinitialized we can use
+        # CLEAR_PARTITIONS_ALL.
+        disk_init_proxy.SetDrivesToClear([])
+        disk_init_proxy.SetDevicesToClear([])
+
+        fresh_disks = [d.name for d in storage.disks if d.partitioned and
+                       not d.format.exists]
+
+        destroy_actions = storage.devicetree.actions.find(
+            action_type="destroy",
+            object_type="device"
+        )
+
+        cleared_partitions = []
+        partial = False
+        for action in destroy_actions:
+            if action.device.type == "partition":
+                if action.device.disk.name not in fresh_disks:
+                    partial = True
+
+                cleared_partitions.append(action.device.name)
+
+        if not destroy_actions:
+            pass
+        elif partial:
+            # make a list of removed partitions
+            disk_init_proxy.SetInitializationMode(CLEAR_PARTITIONS_LIST)
+            disk_init_proxy.SetDevicesToClear(cleared_partitions)
+        else:
+            # if they didn't partially clear any disks, use the shorthand
+            disk_init_proxy.SetInitializationMode(CLEAR_PARTITIONS_ALL)
+            disk_init_proxy.SetDrivesToClear(fresh_disks)
```

Comment:
Could you please move this big `if` part under an abstraction of the separate method?